src/client/renderer/r_model.c

```diff
@@ -153,12 +152,26 @@ r_model_t *R_LoadModel(const char *name) {
 
 		// load the materials first, so that we can resolve surfaces lists
 		R_LoadMaterials(mod);
+		
+		if (mod->type == MOD_BSP) {
+			file_t *file = Fs_OpenRead(file_name);
+
+			// load it
+			format->Load(mod, file);
 
-		// load it
-		format->Load(mod, buf);
+			// close the file
+			Fs_Close(file);
+		} else {
+			void *buf = NULL;
 
-		// free the file
-		Fs_Free(buf);
+			Fs_Load(file_name, &buf);
+
+			// load it
+			format->Load(mod, buf);
+
+			// free the file
+			Fs_Free(buf);
+		}
```

Comment:
Yeah I wasn't sure how to best approach this. I can rework the rest of the loaders to stream the data instead of loading the whole thing into memory maybe?